core/src/main/java/com/linecorp/armeria/common/RequestContext.java

```diff
@@ -435,6 +435,24 @@ default ByteBufAllocator alloc() {
     @MustBeClosed
     SafeCloseable push();
 
+    /**
+     * Immediately run a given {@link Runnable} with this context.
+     */
+    default void run(Runnable runnable) {
+        try (SafeCloseable ignored = push()) {
+            runnable.run();
+        }
+    }
+
+    /**
+     * Immediately call a given {@link Callable} with this context.
+     */
+    default <T> T run(Callable<T> callable) throws Exception {
+        try (SafeCloseable ignored = push()) {
```

Comment:
Could we check if `callable` is `null` using `requireNonNull(...)` before pushing a context?